Framework/Interaction/Tax.php

```diff
@@ -870,4 +882,22 @@ protected function areTimesDifferentDays($timeA, $timeB, $storeId)
 
         return $dateA->format('Y-m-d') != $dateB->format('Y-m-d');
     }
+
+    /**
+     * Return whether product is calculated or not
+     *
+     * @param $item
+     * @return bool
+     */
+    protected function isProductCalculated($item) {
+        if (method_exists($item, 'isChildrenCalculated') && method_exists($item, 'getParentItem')) {
```

Comment:
I'm concerned about defaulting to "false". Are you certain that all product types (simple, configurable, bundled, virtual, downloadable, etc) that are supposed to get taxes have `isChildrenCalculated` _and_ `getParentItem` methods?

Reply:
Never mind. I was thinking that the `$item` variable was a product type instance. After looking at it again, I see that it's an instance of `\Magento\Sales\Model\Order\Item`, and that class will always have the `isChildrenCalculated` and `getParentItem` methods defined.